proto/serve.proto

```diff
@@ -0,0 +1,224 @@
+  // RayServices than this number, the response message will contain a
+  // nextPageToken field you can use to fetch the next page.
+  int32 page_size = 3;
+}
+
+message ListAllRayServicesResponse {
+  repeated RayService services = 1; 
+  // The total number of RayServices for the given query.
+  int32 total_size = 2;
+  // The token to list the next page of RayServices.
+  string next_page_token = 3;
+}
+
+message DeleteRayServiceRequest {
+  // The name of the ray service to be deleted.
+  string name = 1;
+  // The namespace of the ray service to be deleted.
+  string namespace = 2;
+}
+
+message RayService {
+  // Required input field. Unique ray service name provided by user.
+  string name = 1;
+  // Required input field. ray service namespace provided by user
+  string namespace = 2;
+  // Required field. This field indicates the user who owns the ray service.
+  string user = 3;
+  // The config for ray serve deployments.
+  ServeDeploymentGraphSpec serve_deployment_graph_spec = 4;
+  // The cluster template.
+  ClusterSpec cluster_spec = 5;
+  // Output. The status for the ray service.
+  RayServiceStatus ray_service_status = 6;
+  // Output. The time that the ray service created.
+  google.protobuf.Timestamp created_at = 7;
+  // Output. The time that the ray service deleted.     
+  google.protobuf.Timestamp delete_at = 8;
+}
+
+message ServeDeploymentGraphSpec {
+    // Required input field. Import path for load the ray serve.
+    string import_path = 1;
+    // runtime env for loading the ray serve.
+    string runtime_env = 2;
+    // deployments config of the ray serve.
+    repeated ServeConfig serve_configs = 3;
+}
+
+message ServeConfig {
+  // Required input field. Unique ray service deployment name provided by user.
+  string deployment_name = 1;
+  // ray service deployment replicas, default set to 1.  
+  int32 replicas = 2;
+  // ray service deployment route prefix indicated by user.
+  string route_prefix = 3;
+  // max concurrent queries indicate for each ray service deployment.
+  int32 max_concurrent_queries = 4;
+  // user config to the ray serve deployment.
+  string user_config = 5;
+  // auto scaling config for ray serve deployment indicate by user.
+  string autoscaling_config = 6;
+  // actor options indicated resource of each deployment's actor.
+  ActorOptions actor_options = 7;
+}
+
+message ActorOptions {
+  string runtime_env = 1;
+  double cpus = 2;
+  double gpu = 3;
+  int32 memory = 4;
+  int32 object_store_memory = 5;
+  string resource = 6;
+  string acccelerator_type = 7;
+}
+
+message RayServiceStatus {
+  // The ray serve application status.
+  string application_status = 1;
+  // A human-readable description of the status of this operation.
+  string application_message = 2;
+  // The status for each deployment.
+  repeated ServeDeploymentStatus serve_deployment_status = 3;
+  // The related event for the ray service.
+  repeated RayServiceEvent ray_service_event = 4;
```

Comment:
is this event from the serve endpoint?

Reply:
sounds good. BTW, let's name it `ray_service_events`

Reply:
Done.